src/components/backend-ai-registry-list.ts

```diff
@@ -1,66 +1,80 @@
 /**
  @license
- Copyright (c) 2015-2018 Lablup Inc. All rights reserved.
+ Copyright (c) 2015-2022 Lablup Inc. All rights reserved.
  */
 import {get as _text, translate as _t} from 'lit-translate';
 import {css, CSSResultGroup, html, render} from 'lit';
-import {customElement, property} from 'lit/decorators.js';
+import {customElement, query} from 'lit/decorators.js';
 
 import {BackendAIPage} from './backend-ai-page';
+import BackendAIIndicator from './backend-ai-indicator';
+import BackendAIIndicatorPool from './backend-ai-indicator-pool';
 
-import '@vaadin/vaadin-grid/vaadin-grid';
+import './backend-ai-dialog';
+import {default as PainKiller} from './backend-ai-painkiller';
+import {BackendAiStyles} from './backend-ai-general-styles';
+import {IronFlex, IronFlexAlignment} from '../plastics/layout/iron-flex-layout-classes';
 
 import '../plastics/lablup-shields/lablup-shields';
 
-import 'weightless/button';
-import 'weightless/card';
-import 'weightless/icon';
-import 'weightless/label';
-import 'weightless/textfield';
-
 import '@material/mwc-button/mwc-button';
 import '@material/mwc-select/mwc-select';
 import '@material/mwc-list/mwc-list-item';
 import '@material/mwc-switch/mwc-switch';
 
-import './backend-ai-dialog';
-import {default as PainKiller} from './backend-ai-painkiller';
-import {BackendAiStyles} from './backend-ai-general-styles';
-import {IronFlex, IronFlexAlignment} from '../plastics/layout/iron-flex-layout-classes';
+import '@vaadin/vaadin-grid/vaadin-grid';
+
+import 'weightless/button';
+import 'weightless/card';
+import 'weightless/icon';
+import {Label as WlLabel} from 'weightless/label';
+import {Textfield as WlTextfield} from 'weightless/textfield';
 
 /**
  Backend AI Registry List
 
  `backend-ai-registry-list` manages registries.
 
-@group Backend.AI Web UI
- @element backend-ai-release-check
- */
+  @group Backend.AI Web UI
+  @element backend-ai-release-check
+*/
 
 @customElement('backend-ai-registry-list')
 class BackendAIRegistryList extends BackendAIPage {
-  public registryList: any;
-  @property({type: Object}) indicator = Object();
-  @property({type: Number}) selectedIndex = -1;
-  @property({type: Object}) _boundIsEnabledRenderer = this._isEnabledRenderer.bind(this);
-  @property({type: Object}) _boundControlsRenderer = this._controlsRenderer.bind(this);
-  @property({type: Object}) _boundPasswordRenderer = this._passwordRenderer.bind(this);
-  @property({type: Array}) _registryTypes;
-  @property({type: Array}) allowed_registries;
-  @property({type: Array}) hostnames;
-  @property({type: String}) registryType = 'docker';
-  @property({type: Boolean}) editMode;
+  private _allowed_registries: Array<object>;
+  private _editMode: boolean = false;
+  private _hostnames: Array<string>;
+  private _indicator!: BackendAIIndicatorPool;
+  private _registryList: Array<Record<string, unknown>>;
```

Comment:
In that case, we could consider using [`Partial`](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype) type since [the type of value is optional on the server-side.](https://github.com/lablup/backend.ai/blob/141844d383970b7274b9861b625d7f51a6601998/src/ai/backend/common/etcd.py#L300-L371)